packages/nx-sonarqube/src/executors/scan/utils/utils.ts

```diff
@@ -71,18 +93,20 @@ export async function determinePaths(
       sources.push(dep.sourceRoot);
 
       if (dep.testTarget) {
-        if (dep.testTarget.options.coverageDirectory) {
+        const executor: Executor = getExecutor(dep.testTarget.executor);
+        const coverageDirectoryName: CoverageDirectoryName = getCoverageDirectoryName(executor);
+
+        if (dep.testTarget.options?.[coverageDirectoryName]) {
           lcovPaths.push(
             joinPathFragments(
-              dep.testTarget.options.coverageDirectory
+              dep.testTarget.options[coverageDirectoryName]
                 .replace(new RegExp(/'/g), '')
                 .replace(/^(?:\.\.\/)+/, ''),
               'lcov.info'
             )
           );
-        } else if (dep.testTarget.options.jestConfig) {
+        } else if (executor === '@nx/jest:jest' && dep.testTarget.options?.jestConfig) {
```

Comment:
I would expect another else if for vitest executor and config as it can have a [reportsDirectory](https://vitest.dev/guide/coverage#changing-the-default-coverage-folder-location) set